trpcultivate_phenotypes/src/TripalCultivateValidator/ValidatorTraits/FileTypes.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\trpcultivate_phenotypes\TripalCultivateValidator\ValidatorTraits;
+
+/**
+ * Provides setters focused for setting a file types and file media type (MIME) 
+ * supported by the importer, and getter to retrieve the set value.
+ */
+trait FileTypes {
+  /**
+   * The key used by the setter method to create a file types element 
+   * in the context array, as well as the key used by the getter method 
+   * to reference and retrieve the file types element value.
+   * 
+   * @var string
+   */
+  private string $trait_key = 'file_types';
+
+  /**
+   * Sets a file types with the file media type.
+   *
+   * @param array $extensions
+   *   An array of file extensions (file types) the importer expects the file extension of the data file.
+   * 
+   * @return void
+   * 
+   * @throws \Exception
+   *  - Types array is an empty array.
+   *  - Unsupported extension (cannot resolve mime type using the type-mime mapping array).
+   */
+  public function setFileTypes(array $extensions) {
+    
+    // Extensions array must have a element.
+    if (empty($extensions)) {
+      throw new \Exception('The File Types Trait requires an array of file extensions and must not be empty.');
+    }
+    
+    // File extension and file media type (mime type).
+    $file_type_to_mime = [
```

Comment:
This should be a private variable near the top of the class to make it easier to add to.

I'm also still not sure about this being a one-to-one mapping. It might be best to be a extension => array of mime types mapping to support future cases where multiple mime types apply. I don't think this should cause a problem downstream as we are already setting an array of supported mime types... we just use array merge here instead of array push.

Reply:
Implemented as described but as pubic static property